src/format/format.ts

```diff
@@ -57,14 +57,32 @@ export function formatProblems(
   const totalProblems = problems.length;
   problems = problems.filter((m) => !m.ignored);
   const ignoredProblems = totalProblems - problems.length;
+  const formatJSON = format === 'json';
 
   problems = problems
-    .sort((a, b) => severityToNumber(a.severity) - severityToNumber(b.severity))
-    .slice(0, maxProblems);
-
-  if (!totalProblems) return;
-
-  if (format === 'codeframe') {
+    .sort((a, b) => severityToNumber(a.severity) - severityToNumber(b.severity));
+
+  problems = !formatJSON ? problems.slice(0, maxProblems) : problems;
```

Comment:
I would keep this maxProblems slice too. We can set --max-problems to a huge number in workflows build instead of adding different behavior here.

Reply:
if you are sure, we can do that way, but i don't like the idea that we can have invalid JSON output with <...message...> in the workflows if somehow our number is not huge enough